com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/DebugUtility.java

```diff
@@ -73,23 +92,42 @@ public static IDebugSession launch(VirtualMachineManager vmManager, String mainC
      *             when the debuggee was successfully launched, but terminated
      *             with an error before a connection could be established.
      */
-    public static IDebugSession launch(VirtualMachineManager vmManager, String mainClass, String programArguments, String vmArguments, String classPaths,
-            String cwd, String[] envVars) throws IOException, IllegalConnectorArgumentsException, VMStartException {
+    public static IDebugSession launch(VirtualMachineManager vmManager,
+            String mainClass,
+            String programArguments,
+            String vmArguments,
+            String modulePaths,
+            String classPaths,
+            String cwd,
+            String[] envVars)
+            throws IOException, IllegalConnectorArgumentsException, VMStartException {
         List<LaunchingConnector> connectors = vmManager.launchingConnectors();
         LaunchingConnector connector = connectors.get(0);
 
         Map<String, Argument> arguments = connector.defaultArguments();
         arguments.get("suspend").setValue("true");
+
+        String options = "";
         if (StringUtils.isNotBlank(vmArguments)) {
-            arguments.get("options").setValue(vmArguments + " -cp \"" + classPaths + "\"");
-        } else {
-            arguments.get("options").setValue("-cp \"" + classPaths + "\"");
+            options = vmArguments;
+        }
+        if (StringUtils.isNotBlank(modulePaths)) {
+            options += " --module-path \"" + modulePaths + "\"";
+        }
+        if (StringUtils.isNotBlank(classPaths)) {
+            options += " -cp \"" + classPaths + "\"";
```

Comment:
Would suggest StringBuilder for options

Reply:
For simple string concatenation operation, i would prefer the concatenation because the code is shorter. For loop or large string operations, StringBuilder is preferable.